pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -588,6 +588,22 @@ func Convert_v1_FeatureHyperv_To_api_FeatureHyperv(source *v1.FeatureHyperv, hyp
 	return nil
 }
 
+// If Period is lower or equal 0 than we don't add ballooning device. Defaulting to 10 seconds pooling.
```

Comment:
This point isn't clear to me. Specifically, why do we want to silently drop the configuration instead of rejecting the spec?
I think, creating a validating-webhook and rejecting the spec with an error message will make this less unexpected for the users.

Also, please add a few words wrt why values <10 aren't accepted.

Reply:
I'm not too familiar with KubeVirt's API design so apologies if this is out of line, but conflating "period is zero" with "no balloon should be present" doesn't sound like a good idea to me.

I can see "period is zero" mapping quite naturally to "disable stats", but disabling the balloon device altogether should look more along the lines of

```yaml
memBalloon:
  present: false
```

in my opinion.

Reply:
This PR is just enabling the collection of stats and by default, we are disabling the balloon device, right now. Therefore 0 means disabling device... I agree  `present: false` looks nicer to me.

@vladikr I suggest to not reject it as user might don't want to have memballon device. Values <10 are accepted, just <0 values means users don't want to collect the metrics.

Reply:
@xpivarc ok, this is reasonable, but then I wonder why do we want to expose these low level settings to the user at all? Can we just do the right thing? Alternatively, use a cluster wide setting?

Reply:
https://github.com/kubevirt/kubevirt/pull/3568#discussion_r442108913

Reply:
Ok, I see. I probably wouldn't expose such a low level tunable to on the API, but I'm not against it. On the other hand, we shouldn't make the device appear or disappear unexpectedly.
Let's just reject values <0 and introduce an explicit `present: false` as @andreabolognani  suggested.

Reply:
actually, we have examples of how to not add a device:
`VMI.spec.domain.devices.autoAttachGraphics: false` for example will opt-out of adding a graphics adapter (I'm not 100% sure on the field name). We have additional fields opting out of other features as well - see autoAttach* in https://kubevirt.io/api-reference/master/definitions.html#_v1_devices

Thus here we could add `VMI.spec.domain.devices.auttoattachMemoryBallooning: false` to opt out.
WDYT?

Reply:
I was not aware there was prior art for this kind of behavior... I agree that something like `autoattachMemoryBalloon` makes the most sense.

Reply:
Thanks, Fabian for suggestion! I revisited how we actually expose metrics. We collect metrics from virt-launcher only when /metrics endpoint of virt-handler is being scraped(which is cluster-wide setting in Prometheus). Therefore I am going with cluster-wide setting of period in virt-config. Also will add `autoattachMemoryBalloon` as it now makes the most sense(we don't need to have MemBalloon struct on API).